benchmarks/validate_report.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+import os
+
+from utils.report import ACCEPTABLE_METRIC_DEVIATION, METRICS_VALIDATED, Report
+from utils.update_artifacts import (
+    BENCHMARK_ARTIFACTS_PATH,
+    BENCHMARK_REPORT_FILE,
+    BENCHMARK_REPORT_PATH,
+)
+
+
+def metric_valid(key, obs_val, exp_val, threshold):
+    # In case of throughput, higher is better
+    # In case of memory, lower is better.
+    # We ignore lower values for memory related metrices
+    lower = False
+    if key != "throughput":
+        lower = True
+    return check_if_within_threshold(exp_val, obs_val, threshold) or (
+        (obs_val < exp_val and lower)
+    )
+
+
+def check_if_within_threshold(value1, value2, threshold):
+    if float(value1) == 0.0:
+        return True
+    return abs((value1 - value2) / float(value1)) <= threshold
```

Comment:
wondering what is our criteria for threshold?

Reply:
This is a good question. I don't have the correct answer for this. Currently, its at 30% for all metrics. Over time, depending on the failures and the reasons for failures ( could be network issues, or the machine itself being slow), we might have to configure a different threshold for different metrics. I believe this would also help us understand why we dont get consistent behavior across all metrics.